app/environments/brassbirmingham/brassbirmingham/envs/classes/board.py

```diff
@@ -157,26 +178,58 @@ def areNetworked(
     def removeXCoal(self, X: int, towns: List[Town], player: Player):
         for town in towns:
             availableCoal = self.getAvailableCoalBuildingsTradePosts(town)
-            if availableCoal == 0:
+            if len(availableCoal) == 0:
                 continue
 
             _available = availableCoal.pop(
                 0
-            )  # todo assert these changes to resourceAmount are actually affecting the building
```

Comment:
Did we check this TODO?

Reply:
I added asserts in our test file. Now that I'm thinking of it not sure if that's what we want or if we also want assertions done in the code here?

Reply:
Well as long as we’ve confirmed it works correctly, we probably don’t need an assert